### PR TITLE
Enable login

### PR DESCRIPTION
**Pull request by CarlosCRG19**
Basically, added client logic for login process. We are still missing an error handling mechanism, I've been checking this two libraries but couldn't get them to catch axios errors:
* https://www.npmjs.com/package/react-error-boundary
* https://www.npmjs.com/package/react-native-exception-handler

### CODE REVIEW

File: food-bank-app/src/hooks/useClient/httpClient.ts [hunk 1]
```diff
@@ -23,16 +29,50 @@ class HttpClient {
...
+
+    const accessToken: string = await AsyncStorage.getItem('accessToken') ?? ''
+
+    if (requestConfig.headers?.Authorization == null && accessToken != null) {
+      requestConfig.headers = {
+        ...requestConfig.headers,
+        Authorization: `Bearer ${accessToken}`
+      }
+    }
+
```
Comment (CarlosCRG19) [inline food-bank-app/src/hooks/useClient/httpClient.ts:47]: this basically adds the access token before every request, which will work for the rest of the calls after login... this is an async operation tho an I have not really tried it.

On web it is most easier as localStorage is synchronous... I've been thinking about using this package:
https://www.npmjs.com/package/sync-storage

Comment (antonieto) [reply food-bank-app/src/hooks/useClient/httpClient.ts:47]: Have you checked [expo-secure-store]( https://docs.expo.dev/versions/latest/sdk/securestore/)? it's supposed to be encrypted and thus _safer_.

File: food-bank-app/src/hooks/useClient/httpClient.ts [hunk 2]
```diff
@@ -1,4 +1,10 @@
-import axios, { AxiosInstance, AxiosResponse } from 'axios'
+import axios, {
+  AxiosError,
+  AxiosInstance,
+  AxiosRequestConfig,
+  AxiosResponse
+} from 'axios'
+import AsyncStorage from '@react-native-async-storage/async-storage'
 
 declare module 'axios' {
```
Comment (antonieto) [inline food-bank-app/src/hooks/useClient/httpClient.ts:9]: This seems pretty advanced. Could you briefly explain what this does?

File: food-bank-app/src/hooks/useClient/httpClient.ts [hunk 3]
```diff
@@ -23,16 +29,50 @@ class HttpClient {
...
+    }
+
+    return requestConfig
   }
 
   private readonly _handleResponse = ({ data }: AxiosResponse): any => data
 
+  private readonly _handleError = async (error: AxiosError): Promise<any> => {
+    if (process.env.TARGET_ENV !== 'production') {
+      // This is temporary, WE REALLY NEED AN ERROR HANDLING MECHANISM
```
Comment (antonieto) [inline food-bank-app/src/hooks/useClient/httpClient.ts:55]: +1

**Review by antonieto — APPROVED** [review]
lgtm

**Review by BrandonMagana — APPROVED** [review]
Awesome work @CarlosCRG19 :)

**Pull request merged**